ballerina/endpoint.bal

```diff
@@ -23,16 +23,26 @@ import ballerina/jballerina.java;
 public isolated client class Client {
     private ConnectionConfig clientEndpointConfig = {};
 
-    final handle datasource;
-
     # Initializes the connector. During initialization you have to pass needed configurations such as host name, port number and password.
     #
     # + config - Configuration for the connector
     # + return - `http:Error` in case of failure to initialize or `null` if successfully initialized
-    public isolated function init(ConnectionConfig config) returns error? {
-        self.datasource = initClient(config);
+    public isolated function init(ConnectionConfig config) returns Error? {
+        Error? initClientResult = self.initClient(self, config);
```

Comment:
Can't we do this ?
```suggestion
        check self.initClient(self, config);
```

Reply:
modified with https://github.com/ballerina-platform/module-ballerinax-redis/pull/197/commits/6d826c80794537dc41bc07fc1bf660473cc4c8c5